tests/migration_test.go

```diff
@@ -1835,6 +1835,114 @@ var _ = Describe("[Serial][rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][leve
+					if strings.HasPrefix(key, v1.CPUModelLabel) && strings.Contains(key, model) {
+						return true
+					}
+				}
+				return false
+			}
+			expectFeatureToBeSupportedOnNode := func(node *k8sv1.Node, features []string) {
+				isFeatureSupported := func(feature string) bool {
+					for key, _ := range node.Labels {
+						if strings.HasPrefix(key, v1.CPUFeatureLabel) && strings.Contains(key, feature) {
+							return true
+						}
+					}
+					return false
+				}
+
+				supportedFeatures := make(map[string]bool)
+				for _, feature := range features {
+					supportedFeatures[feature] = isFeatureSupported(feature)
+				}
+
+				Expect(supportedFeatures).Should(Not(ContainElement(false)))
+			}
+			getOtherNodes := func(nodeList *k8sv1.NodeList, node *k8sv1.Node) (others []*k8sv1.Node) {
+				for _, curNode := range nodeList.Items {
+					if curNode.Name != node.Name {
+						others = append(others, &curNode)
+					}
+				}
+				return others
+			}
+			isHeterogeneousCluster := func() bool {
+				nodes := tests.GetAllSchedulableNodes(virtClient)
+				for _, node := range nodes.Items {
+					hostModel := getNodeHostModel(&node)
+					otherNodes := getOtherNodes(nodes, &node)
+
+					foundSupportedNode := false
+					foundUnsupportedNode := false
+					for _, otherNode := range otherNodes {
+						if isModelSupportedOnNode(otherNode, hostModel) {
+							foundSupportedNode = true
+						} else {
+							foundUnsupportedNode = true
+						}
+
+						if foundSupportedNode && foundUnsupportedNode {
+							return true
+						}
+					}
+				}
+
+				return false
+			}
+
+			It("should migrate only to nodes supporting right cpu model", func() {
+				if !isHeterogeneousCluster() {
+					Skip("to perform test there needs to be a node that can migrate only to some of the nodes on the cluster.")
+				}
+
+				vmi := tests.NewRandomFedoraVMIWithGuestAgent()
+				vmi.Spec.Domain.CPU.Model = v1.CPUModeHostModel
+				vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("1Gi")
+
+				By("Starting the VirtualMachineInstance")
+				vmi = runVMIAndExpectLaunch(vmi, 240)
+
+				originalNode, err := virtClient.CoreV1().Nodes().Get(context.Background(), vmi.Status.NodeName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				hostModel := getNodeHostModel(originalNode)
+				requiredFeatures := getNodeHostRequiredFeatures(originalNode)
+
+				runStressTest(vmi, "800", stressdefaultTimeout)
+
+				// execute a migration, wait for finalized state
+				By("Starting the Migration")
+				migration := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+				_ = tests.RunMigrationAndExpectCompletion(virtClient, migration, 180)
+
+				newNode, err := virtClient.CoreV1().Nodes().Get(context.Background(), vmi.Status.NodeName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(isModelSupportedOnNode(newNode, hostModel)).To(BeTrue())
+				expectFeatureToBeSupportedOnNode(newNode, requiredFeatures)
```

Comment:
In order to guarantee that at least the expected scheduling occurs for both heterogeneous and homogeneous clusters,  you could expand this verification to include cross examining the pod to ensure the right affinity/selectors are set. That would tell us that at least from a scheduling stand point that the host mode migration worked as expected.

Reply:
You're absolutely right.
The test is now updated and will occur whether we're running on a homogeneous cluster or not. The host model & node-selector labels will be check for original/target node/pod.